tests/auditor/test_data/test_valid_rules1.yaml

```diff
@@ -0,0 +1,47 @@
+# Copyright 2017 The Forseti Security Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Test rule configurations
+# Valid rules
+
+rules:
```

Comment:
What if we restructured it like this?

variables:
rules:
-rule_id:
 description
 input_filter:
  type: <aggregate, individual, ....>
  resource_type:
  match: <what it's matching>
 condition:

It could generically handle getting the specific resources, filtering, and then aggregating or passing individually. Then a rule could be:
variables:
 - golden_policy: [<policies>]
-rule_id: Matches
 description
 input_filter:
  type: aggregate
  resource_type: firewall_rule
  match: *
 condition: input == golden_policy

Reply:
Let's discuss on Monday during the meeting so we can formalize what we want the generic rule grammar to be like and how we can apply it to more specific cases.